app/models/grda_warehouse/client_matcher_lookups/ssn_lookup.rb

```diff
@@ -0,0 +1,43 @@
+###
+# Copyright 2016 - 2024 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+module GrdaWarehouse::ClientMatcherLookups
+  class SSNLookup < BaseLookup
+    attr_accessor :format
+    def initialize(format: :full)
+      raise ArgumentError, "unknown format: #{format}" unless format.in?([:full, :last_four])
+
+      self.format = format
+      super()
+    end
+
+    def get_ids(ssn:)
+      return [] unless valid_ssn?(ssn)
+
+      @values[ssn]&.uniq || []
+    end
+
+    def add(client)
+      key = client.ssn
+      key = key[-4..] if key && format == :last_four
```

Comment:
Any reason not to use?
```suggestion
      key = key.last(4) if key && format == :last_four
```